x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/timing_footers/timing_footers_context.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { createContext, FunctionComponent, ReactNode, useContext, useMemo } from 'react';
+import { UseField, useFormData } from '../../../../../shared_imports';
+import { FormInternal } from '../../types';
+import { calculateRelativeTimingMs, normalizeTimingsToHumanReadable } from '../../lib';
+import { TimingFooter } from './timing_footer';
+
+export interface TimingFooters {
+  hot: ReactNode;
+  warm: ReactNode;
+  cold: ReactNode;
+  setDeletePhaseEnabled: (value: boolean) => void;
+}
+
+const TimingFootersContext = createContext<TimingFooters>(null as any);
+
+export const TimingFootersProvider: FunctionComponent = ({ children }) => {
+  const [formData] = useFormData<FormInternal>();
+
+  const phaseTimings = useMemo(() => {
+    return calculateRelativeTimingMs(formData);
+  }, [formData]);
+
+  const humanReadableTimings = useMemo(() => normalizeTimingsToHumanReadable(phaseTimings), [
+    phaseTimings,
+  ]);
```

Comment:
Currently these functions (calculateRelativeTimingMs, and normalizeTimingsToHumanReadable) are being run twice, once here and once in the timeline.tsx component - for the same data. I wonder if it might be worth creating a separate context for this where all components can share the output of these computations. No need to action any of this here, we can do it as a follow up PR. I have some ongoing work to refactor some of the timeline component and timing computation logic: https://github.com/elastic/kibana/pull/89422